portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -583,6 +583,25 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	'getGeneticProfileIds': function () {
 	    return this.genetic_profile_ids;
 	},
+	'getDefaultGeneticProfileId': function () {
+			var gp_ids = this.getGeneticProfileIds();
```

Comment:
There's a method `getGeneticProfiles` on the data manager which includes other metadata about the genetic profiles besides the ids, including the data type. You should use that instead of using `indexOf` on the id.